### PR TITLE
feat(platform): update PR target branch if baseBranch changed

### LINKED ISSUE #19900: update pr upon changing baseBranch
### What would you like Renovate to be able to do?

as a renovate user
in order to change the baseBranch without having to update myself the existing PRs targetting an old base branch
I need renovate to update existing PRs baseBranch when renovate baseBranch is changed

Workaround: update the PRs using the script below using [github cli](https://cli.github.com/)

```bash
#!/bin/bash

set -x
set -f # disable file name expansion, especially in for loops

FROM="feature-bumps-for-v55"
TARGET="feature-bumps"
LIMIT="300"

echo "Editing renovate PRs to change base branch from ${FROM} to ${TARGET}"

PRS=$(gh pr list \
  --limit ${LIMIT} \
  --app "renovate" \
  --base "${FROM}" \
  --json number \
  | jq -r '.[].number' )

echo "first PRs (maxed at ${LIMIT}} to edit are: ${PRS}"

for p in $PRS; do
  gh pr edit $p --base ${TARGET}
done;


echo "rereun this command to check not more PR beyond $LIMIT is left to process"
```

### If you have any ideas on how this should be implemented, please tell us here.

similar as current PR submissions

### Is this a feature you are interested in implementing yourself?

No

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes
Sets destination branch when updating existing pull requests.

## Context
Closes: https://github.com/renovatebot/renovate/issues/19900

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/platform/bitbucket/index.spec.ts [hunk 1]
```diff
@@ -1428,6 +1428,25 @@ describe('modules/platform/bitbucket/index', () => {
...
 
+    it('updates target branch of the PR', async () => {
+      const scope = await initRepoMock();
+      scope
+        .get('/2.0/repositories/some/repo/pullrequests/5')
+        .reply(200, { values: [pr] })
+        .put('/2.0/repositories/some/repo/pullrequests/5')
+        .reply(200)
+        .post('/2.0/repositories/some/repo/pullrequests/5/decline')
+        .reply(200);
```
Comment (RahulGautamSingh) [inline lib/modules/platform/bitbucket/index.spec.ts:1439]: wrong

**Review by rarkins — COMMENTED** [review]
So this PR means we always set the target branch when updating a PR, even if the branch hasn't changed?

**Comment by RahulGautamSingh** [comment]
> So this PR means we always set the target branch when updating a PR, even if the branch hasn't changed?

Yes

**Comment by rarkins** [comment]
Tested on each platform except Bitbucket server?

**Comment by RahulGautamSingh** [comment]
> Tested on each platform except Bitbucket server?

codecommit is pending too.

**Comment by rarkins** [comment]
So is this still in progress, or ready to merge?

**Comment by RahulGautamSingh** [comment]
> So is this still in progress, or ready to merge?

I am stuck with codecommit as there isn't  any command to update the destination-branch of an existing PR using the [@aws-codecommit0client docs](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-codecommit/classes/codecommit.html#updatepullrequesttitle) we are using.

**Comment by rarkins** [comment]
That's what I thought. Let's skip it

**Review by rarkins — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**